### PR TITLE
[On Device Quantization][pytorch]Make insert_quant_dequant support ondevice ptq

### PR DESCRIPTION
**Pull request by kimishpatel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #84131
* #83807
* #83742
* #83571
* __->__ #83570
* #83569
* #83568

Summary:
This diff adds a way to:
- clone previously observed method
- Add calls to observer's calculate_qparams methods
- Extract the scale and zero point
- Use them to insert quant dequant nodes

Now for forward method we have
- observe_forward
- quantize_forward

observe_forward is used post training to observer statistics. In the
case of dynamic PTQ this requires just running that method once to
update weight observer statistics.

quantize_forward method will be used to use the observer
statistics to calculate quantization parameters and apply that to quant
dequant op.

Subsequent diffs will replace dequant + op with their quantized op
counter parts and replace quantize ops with relevant packed params class
where possible

Test Plan:
To be written

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D38771419](https://our.internmc.facebook.com/intern/diff/D38771419)

### CODE REVIEW

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38771419).

File: test/quantization/jit/test_ondevice_quantization.py [hunk 1]
```diff
@@ -10,8 +10,13 @@
 
 from torch.ao.quantization.quantize_jit import (
     prepare_ondevice_dynamic_jit,
+    convert_ondevice_dynamic_jit,
 )
 
+from torch.ao.quantization.quant_type import QuantType
```
Comment (jerryzh168) [inline test/quantization/jit/test_ondevice_quantization.py:16]: +1 lint

**Review by jerryzh168 — APPROVED** [review]

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38771419).

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38771419).

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38771419).

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38771419).

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38771419).

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38771419).

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38771419).

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38771419).

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38771419).

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38771419).

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38771419).

**Pull request closed without merging**